### DISCUSSION TITLE
Most time gui spends in `femtovg::path::cache::PathCache::tesselate_bezier`, is this optimal?

### DISCUSSION THREAD

**Discussion by qarmin** [slint-ui/slint#4109] [Q&A]
When testing my GTK app I found that a lot of time app spend in different libraries like librsvg, pango etc.

I wanted to check how much time spend app with slint frontend and I found that most of ther time, slint spends in femtovg tesselate_bezier function.

App is compiled with release profile

Is this really needed to call it so much time? - it is executed few times per second

This is not really problem for me, since app works quite smooth on my PC(but I use old i7 processor), so maybe performance problems may be visible on slower computers 

https://github.com/slint-ui/slint/assets/41945903/2b04a350-481e-4c96-9532-1a261f30c1b3

With software renderer I see that CPU usage is distributed in several functions and I don't see anything similar to femotvg function

https://github.com/slint-ui/slint/assets/41945903/d825be0f-2f64-43de-ab03-eab2ff6698d1

With skia also I don't see similar thing

https://github.com/slint-ui/slint/assets/41945903/921dbf7d-3b72-4bbd-9976-a27b4ca580d7

When counting only app code, this function takes ~50% of time

Also reported to https://github.com/femtovg/femtovg/discussions/186

**Comment by tronical** [comment]
Thanks again for your detailed report. You're poking at the right issue :). I believe the other renderers (Qt, Skia) have caching going on to avoid re-doing the tessellation. Plus I guess the elements rendered here, likely border rectangles, are drawn using a dedicated code path perhaps, while with FemtoVG it's the general path renderer. There's a path cache in FemtoVG as well, but it's dependent on the global transformation. So once you start moving a Rectangle it starts falling apart (due to resize or scrolling).

As you point out in https://github.com/femtovg/femtovg/discussions/186, perhaps there's a way to also make the function itself faster in FemtoVG - but I don't know that code path well enough to make that call :(
